src/Symfony/Component/Security/Http/EventListener/IsGrantedAttributeListener.php

```diff
@@ -65,8 +65,10 @@ public function onKernelControllerArguments(ControllerArgumentsEvent $event)
             if (!$this->authChecker->isGranted($attribute->attribute, $subject)) {
                 $message = $attribute->message ?: sprintf('Access Denied by #[IsGranted(%s)] on controller', $this->getIsGrantedString($attribute));
 
-                if ($statusCode = $attribute->statusCode) {
-                    throw new HttpException($statusCode, $message);
+                $statusCode = $attribute->statusCode;
+                $errorCode = $attribute->errorCode;
+                if ($statusCode || $errorCode) {
+                    throw new HttpException($statusCode ?? 403, $message, null, [], $errorCode ?? 0);
                 }
 
                 $accessDeniedException = new AccessDeniedException($message);
```

Comment:
somewhat unrelated, but i'd prefer always throwing HttpException or always throwing AccessDeniedException

Reply:
I feel that a decision like this is out of scope of this PR, but to me it would make sense to allow for a more generic customizable exception here in favor of the AccessDeniedException. Either that, or go strictly for the AccessDeniedException and make the exception code a parameter there.